department-of-veterans-affairs/vets-api
# [VI-252] MAP SecurityToken duration validation

## Summary

- Updates MAP STS service to validate the expiration time of the returned token; tokens valid for more than `900` seconds are considered invalid and will result in a `502 bad gateway` error if requested through the MAP services controller.

## Related issue(s)

- https://jira.devops.va.gov/browse/VI-252

## Testing done

- Create an STS token for a MAP token request & make the request: `locahost:3000/v0/map_services/chatbot/token` - you should receive a 200 response & a token
- Testing behavior on localhost is controlled through vets-api-mockdata file - `vets-api-mockdata/map/secure_token_service/token.yml`
- Update the mockdata `expires_in` value to greater than 900 & re-request the token - you should receive a 502 Bad Gateway error

## Screenshots
![image](https://github.com/user-attachments/assets/0044cdd5-27d2-4262-9536-44a989a7eb74)

## What areas of the site does it impact?
MAP STS requests

## Acceptance criteria


## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_
